service/timestreamwrite/api_client.go

```diff
@@ -87,6 +97,16 @@ type Options struct {
 	// The HTTP client to invoke API calls with. Defaults to client's default HTTP
 	// implementation if nil.
 	HTTPClient HTTPClient
+
+	// The cache used to store discoevred endpoint. Defaults to 10.
+	CacheSize int64
+
+	// EndpointUsedForDiscovery is the endpoint used internally by the operation
+	// with the endpoint discovery api request call.
+	EndpointUsedForDiscovery string
+
+	// EnableEndpointDiscovery indicates if endpoint discovery is enabled, auto or disabled.
+	EnableEndpointDiscovery config.EndpointDiscoveryEnableState
```

Comment:
We should consider making these a single interface member value that is auto populated by the client's constructor if unset by the user. 

This would allow the endpoint discovery interface to be mocked out or replaced similar to `HTTPSigner` and `HTTPClient`.